### ISSUE TITLE
Block Tracker MEGA report - Y's, Chests and Gamemodes

### ISSUE THREAD

**Issue by ChiefArug** [TeamPneumatic/pnc-repressurized#1053]
### Describe the bug

The block tracker upgrade does not track blocks above y 255 or below y 0
![up](https://user-images.githubusercontent.com/73862885/180218788-5c53d122-02f7-4cd2-b78a-cbd0b5cdee4b.png)
![down](https://user-images.githubusercontent.com/73862885/180218797-e3576de9-0a89-46fb-9f54-30095e977d62.png)

The block tracker will generate loot in chests when you look at them. It does not do this for PNC's chests though, or AE2's sky stone chest, making me think its something to do with modded ones.
It will also generate loot into 'locked' chests, which normally cannot be opened (and therefore generated) until you are holding an item with the same name as the 'lock'.
Furthermore it will generate the loot even when the wearer is in spectator mode.

### How to reproduce the bug

Command for upgraded helmet with block tracker and speed upgrades: 
```
/give @s pneumaticcraft:pneumatic_helmet{Damage:0,UpgradeInventory:{Items:[{Count:1b,Slot:1,id:"pneumaticcraft:block_tracker_upgrade"},{Count:10b,Slot:2,id:"pneumaticcraft:speed_upgrade"}],Size:9},"pneumaticcraft:air":120000}
```
Place a block with an inv above 255 or below 0, and look at it with the block tracker upgrade.

Run each of these commands, then look at them with the block tracker, the vanilla ones will generate but the modded ones will not. Doing it in spectator has identical results.
```
/setblock ~ ~ ~ chest{LootTable:"minecraft:chests/end_city_treasure"}
/setblock ~ ~ ~ pneumaticcraft:reinforced_chest{LootTable:"minecraft:chests/end_city_treasure"}
/setblock ~ ~ ~ ae2:sky_stone_chest{Lock:"yes",LootTable:"minecraft:chests/end_city_treasure"}
/setblock ~ ~ ~ chest{Lock:"yes",LootTable:"minecraft:chests/end_city_treasure"}
```
The last one is for a locked chest that is unopenable unless you hold an item with the same name as the lock. It should not generate unless you are holding said item, like vanilla. 
Command for obtaining a stick that can open aforementioned chest
```
/give @s minecraft:stick{display:{Name:'{"text":"yes"}'}}
```

### Expected behavior

It to work

It to not generate stuff in spectator mode, and for it to generate loot chests from mods. It to not generate locked chests unless holding an item that can unlock it (or not at all, if thats easier)

### Which Minecraft version are you using?

1.18

### Which version of PneumaticCraft: Repressurized are you using?

pneumaticcraft-repressurized-1.18.2-3.3.1-110

**Comment by desht** [comment]
* Y problem is a super easy fix
* Yeah the block tracker only looks for vanilla blocks with a loot table. I can add PNC chests easily enough, but there's no mod-independent way of knowing if a chest (or indeed any block entity) has a loot table...
* Lockable chests shouldn't be a big problem to support

**Comment by desht** [comment]
Fixed in 3.4.0 release
